src/lib.rs

```diff
@@ -338,13 +326,12 @@ pub fn generate_flamegraph_for_workload(workload: Workload, opts: Options) -> an
         signal_hook::low_level::register(SIGINT, || {}).expect("cannot register signal handler")
     };
 
-    let use_sudo = opts.root.is_some();
     let perf_output = if let Workload::ReadPerf(perf_file) = workload {
         Some(perf_file)
     } else {
         arch::initial_command(
             workload,
-            opts.root,
+            opts.root.as_ref(),
```

Comment:
Because we'll need a more complicated transformation here, let's extract it at the point where `use_sudo` was previously defined so we can pass the output to both `initial_command()` and `output()`?